### PR TITLE
Fix package command for complex Dockerfiles

### PR DESCRIPTION
**Pull request by munkyboy**

### CODE REVIEW

**Comment by munkyboy** [comment]
bah, my `--pull` change breaks this test/use-case:

```
  Scenario: with a docker artifact based on a local image
    Given a file named "Dockerfile" with:
    """
    FROM elsyblackbox_docker_artifact
    """
    And a file named "lc.yml" with:
    """yaml
    name: testpackage
    """
    When I run `lc package --docker-image-name=elsyblackbox_docker_artifact`
    Then it should succeed with "Successfully built "
```

I think relying on local images like this is an anti-pattern and we should drop support for it. Thoughts?

**Comment by paulcichonski** [comment]
I can't remember why we even had that test?

File: command/package.go [hunk 1]
```diff
@@ -63,21 +63,21 @@ func RunPackage(c *cli.Context) error {
...
-		}
-
 		if len(dockerImageName) == 0 {
 			logrus.Panic("you must use `--docker-image-name` to package a docker image")
 		}
 
 		buildArgs := []string{"build", "-t", dockerImageName}
+
+		if !c.GlobalBool("offline") {
+			buildArgs = append(buildArgs, "--pull")
```
Comment (paulcichonski) [inline command/package.go]: is this major version bump territory?

File: blackbox-test/features/package.feature [hunk 2]
```diff
@@ -138,6 +152,38 @@ Feature: package task
...
+    Then it should succeed
+    And the output should contain all of these:
+     | Image is up to date for alpine:latest  |
+     | Image is up to date for busybox:latest |
+
+   Scenario: with build docker artifact which derives image from a build-arg
+     Given a file named "Dockerfile" with:
+     """
+     ARG b=library/alpine
+     FROM ${b}
```
Comment (paulcichonski) [inline blackbox-test/features/package.feature]: does this not work without the `build --pull`?

**Comment by munkyboy** [comment]
fixed up the PR to explicitly remove support for building from local images

**Comment by paulcichonski** [comment]
If we are removing support for something the version bump should be 5.0.0.

**Comment by paulcichonski** [comment]
ah, right, but the commit message is still "release v4.3.0"

**Comment by munkyboy** [comment]
yeah, I plan to rewrite when I squash

**Pull request merged**